R/settings_utils.R

```diff
@@ -61,44 +68,41 @@ find_config <- function(filename) {
     dirname(filename)
   }
 
+  path <- normalizePath(path, mustWork = FALSE)
+
   # NB: This vector specifies a priority order for where to find the configs,
   # i.e. the first location where a config exists is chosen and configs which
   # may exist in subsequent directories are ignored
   file_locations <- c(
     # Local (incl. parent) directories
-    find_config2(path),
+    find_local_config(path, basename(linter_file)),
     # User directory
     # cf: rstudio@bc9b6a5 SessionRSConnect.R#L32
     file.path(Sys.getenv("HOME", unset = "~"), linter_file),
     # Next check for a global config file
     file.path(R_user_dir("lintr", which = "config"), "config")
   )
 
-  # Search through locations, return first valid result
-  for (loc in file_locations) {
-    if (file.exists(loc)) {
-      return(loc)
-    }
-  }
-
-  NULL
+  first_exists(file_locations)
 }
 
-find_config2 <- function(path) {
-  config <- basename(getOption("lintr.linter_file"))
-  path <- normalizePath(path, mustWork = FALSE)
-
-  while (!has_config(path, config)) {
-    gh <- file.path(path, ".github", "linters")
-    if (has_config(gh, config)) {
-      return(file.path(gh, config))
+find_local_config <- function(path, config_file) {
+  repeat {
+    guesses_in_dir <- c(
+      file.path(path, config_file),
+      file.path(path, paste0(config_file, ".R")),
```

Comment:
Swapping the order allows for smoother testing (no need to delete the `.lintr` file for testing and a rollback can be achieved by deleting `.lintr.R`)

Reply:
not sure I follow you here, sorry

Reply:
`first_exists()` returns the first of the files that exist.
If a `.lintr` file exists, it will be returned even if `.lintr.R` exists.

I imagine noone will have a `.lintr.R` lying around.
Now if you want to test the feature, you need to rename your `.lintr` to some other file name; otherwise `find_config()` will still pick up the DCF file.

Reply:
OK, changed the precedence